core/templates/pages/exploration-editor-page/feedback-tab/feedback-tab.component.ts

```diff
@@ -16,245 +16,254 @@
-            result.newThreadSubject, result.newThreadText)
-        ).then(() => {
-          ctrl.clearActiveThread();
-          AlertsService.addSuccessMessage('Feedback thread created.');
-        },
-        () => {
-          // Note to developers:
-          // This callback is triggered when the Cancel button is clicked.
-          // No further action is needed.
-        });
-      };
-
-      var _isSuggestionHandled = () => {
-        return (
-          ctrl.activeThread !== null &&
-          ctrl.activeThread.isSuggestionHandled());
-      };
-
-      var _isSuggestionValid = () => {
-        return (
-          ctrl.activeThread !== null &&
-          ExplorationStatesService.hasState(
-            ctrl.activeThread.getSuggestionStateName()));
-      };
-
-      var _hasUnsavedChanges = () => {
-        return ChangeListService.getChangeList().length > 0;
-      };
-
-      ctrl.getSuggestionButtonType = () => {
-        return (
-          !_isSuggestionHandled() && _isSuggestionValid() &&
-          !_hasUnsavedChanges()) ? 'primary' : 'default';
-      };
-
-      // TODO(Allan): Implement ability to edit suggestions before applying.
-      ctrl.showSuggestionModal = () => {
-        if (ctrl.activeThread === null) {
-          throw new Error(
-            'Trying to show suggestion of a non-existent thread');
-        }
-        SuggestionModalForExplorationEditorService.showSuggestionModal(
-          ctrl.activeThread.suggestion.suggestionType, {
-            activeThread: ctrl.activeThread,
-            setActiveThread: (
-              threadId => ctrl.fetchUpdatedThreads()
-                .then(() => ctrl.setActiveThread(threadId))),
-            isSuggestionHandled: _isSuggestionHandled,
-            hasUnsavedChanges: _hasUnsavedChanges,
-            isSuggestionValid: _isSuggestionValid
-          }
-        );
-      };
-
-      ctrl.addNewMessage = function(threadId, tmpText, tmpStatus) {
-        if (threadId === null) {
-          AlertsService.addWarning(
-            'Cannot add message to thread with ID: null.');
-          return;
-        }
-        if (!tmpStatus) {
-          AlertsService.addWarning('Invalid message status: ' + tmpStatus);
-          return;
-        }
-        ctrl.threadIsStale = true;
-        ctrl.messageSendingInProgress = true;
-        let thread = ThreadDataBackendApiService.getThread(threadId);
-        if (thread === null) {
-          throw new Error(
-            'Trying to add message to a non-existent thread.');
+@Component({
+  selector: 'oppia-feedback-tab',
+  templateUrl: './feedback-tab.component.html'
+})
+export class FeedbackTabComponent implements OnInit, OnDestroy {
+  directiveSubscriptions = new Subscription();
+  STATUS_CHOICES = this.threadStatusDisplayService.STATUS_CHOICES;
+  activeThread: SuggestionThread;
+  userIsLoggedIn: boolean;
+  threadIsStale: boolean;
+  threadData: FeedbackThread[];
+  messageSendingInProgress: boolean;
+  tmpMessage: {
```

Comment:
```suggestion
  tmpMessage: {
```
This variable name be more declarative.

Reply:
updated